DDD.Core/AzureStorage/TableStorageRepository.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace DDD.Core.AzureStorage
+{
+    public interface ITableStorageRepository<T> where T : class, ITableEntity, new()
+    {
+        Task InitializeAsync();
```

Comment:
This feels like an odd method to have on the public interface. I generally try to avoid two phase setup as makes consumption more complicated. I would consider encapsulating the call to check the table is created within the implementation itself.

Reply:
Changing the name to something clearer would help I feel. CreateTableIfNotExistsAsync is actually quite a good name. Helpful error message to highlight invalid usage would also be good.

Reply:
I agree, but if I want to keep it async then it has to be a separate method since C# doesn't allow async ctors. I didn't really want to bleed the implementation detail of creating a table into the repo if that makes sense?